src/test/java/seedu/sudohr/model/department/DepartmentTest.java

```diff
@@ -19,6 +22,50 @@ public void asObservableList_modifyList_throwsUnsupportedOperationException() {
         assertThrows(UnsupportedOperationException.class, () -> department.getEmployees().remove(0));
     }
 
+    @Test
+    public void execute_addEmployee_success() {
+        Department department = new DepartmentBuilder().build();
+
+        department.addEmployee(TypicalEmployees.ALICE);
+        assertTrue(department.hasEmployee(TypicalEmployees.ALICE));
+    }
+
+    @Test
+    public void execute_addDuplicateEmployee_throwsException() {
+        Department department = new DepartmentBuilder().build();
+
+        department.addEmployee(new EmployeeBuilder().build());
+
+        assertThrows(DuplicateEmployeeException.class, () -> department.addEmployee(new EmployeeBuilder().build()));
+    }
+
+    @Test
+    public void execute_addExactDuplicateEmployee_throwsException() {
```

Comment:
For more rigorous testing,
can consider:
addDuplicateEmployee (same reference/object)
addDuplicateEmail
addDuplicatePhoneNumber
addDuplicateId (add 2 employees with different IDs, tho at this stage, this cannot possibly happen ah, so not as important)